src/components/todo-list/todo-list.jsx

```diff
@@ -0,0 +1,66 @@
+import {useState, useEffect} from "react";
+import styled from './todo-list.module.scss';
+import data from '../../../data/index.json';
+import useSearch from './useSearch.js'
+
+const TodoList = () => {
+  const [task, setTask] = useState(data.tasks);
+  const [state, setState] = useState(task.map((e) => e.isCompleted));
+  const [request, setRequest] = useState('')
+  const onCompleteTask = (element, index) => {
+    setState(state.map((e,i) => i === index ? e = element : e));
+  }
+  const onDeleteTask = (index) => {
+    if (task[index] === index) {
+      setTask(task.splice(index+1,1));
+    }
+
+    let item = document.querySelector(`#item-${index+1}`);
+    item.remove();
+  }
+
+  //поиск
+
+  const getSearchRequest = (event) => {
+    setRequest(event.target.value)
+  }
+
+  const onSearch = useSearch(request, task, setTask);
+
+  const onRenderSearchResult = () => {
+    onSearch.onClick()
+  }
+
+  return (
+      <div className={styled.TodoList}>
+        <div className={styled.TodoList__header}>
+          <h1 className={styled.TodoList__headerName}>{data.title}</h1>
+          <input type="text" className={styled.TodoList__headerSearch} placeholder={'Поиск'} onChange={(event) => getSearchRequest(event)} value={onSearch.value}/>
+          <span className={styled.TodoList__headerIcon} onClick={onRenderSearchResult}>
+            <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none">
+              <circle cx="11" cy="11" r="6.25" stroke="#007FFF" strokeWidth="1.5"/>
+              <path d="M15.9058 16.021L20.1944 19.697" stroke="#007FFF" strokeWidth="1.5" strokeLinecap="round"/>
+            </svg>
+          </span>
+        </div>
```

Comment:
Неточность в семантике
Сейчас для начала поиска требуется ввести строку в поле и кликнуть на элемент с иконкой. Другие методы взаимодействия работать не будут
```suggestion
        <form className={styled.TodoList__header} onSubmit=(() => e.preventDefault())>
          <h1 className={styled.TodoList__headerName}>{data.title}</h1>
          <input type="text" className={styled.TodoList__headerSearch} placeholder={'Поиск'} onChange={(event) => getSearchRequest(event)} value={onSearch.value}/>
          <button className={styled.TodoList__headerIcon} onClick={onRenderSearchResult}>
            <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none">
              <circle cx="11" cy="11" r="6.25" stroke="#007FFF" strokeWidth="1.5"/>
              <path d="M15.9058 16.021L20.1944 19.697" stroke="#007FFF" strokeWidth="1.5" strokeLinecap="round"/>
            </svg>
          </button>
        </form>
```
Если использовать теги `form` и `button`, то браузер будет способен автоматически распознавать нажатие клавиши `Enter` в качестве клика по кнопке внутри формы